src/Mutator/Arithmetic/AssignmentEqual.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Arithmetic;
+
+use Infection\Mutator\Mutator;
+use PhpParser\Node;
+
+class AssignmentEqual extends Mutator
+{
+    /**
+     * Replaces "==" with a "=".
+     *
+     * @param Node $node
+     *
+     * @return Node\Expr\Assign
+     */
+    public function mutate(Node $node)
+    {
+        return new Node\Expr\Assign($node->left, $node->right, $node->getAttributes());
+    }
+
+    public function shouldMutate(Node $node): bool
+    {
+        return $node instanceof Node\Expr\BinaryOp\Equal && $node->left instanceof Node\Expr\Variable;
```

Comment:
does that work if you have `$x->getFoo() == $a`? `Variable` doesn't look wide enough for it. Maybe we don't want to cover this case tbh, but I would find it interesting as this is where (im)mutability can important

Reply:
`$x->getFoo() = $a` will always error

Reply:
Here you go: 1f19d3a

Reply:
🤔 yeah you're right. Now sure where I saw something like that I must be confusing with something else